csharp-intermediate-topics/WhenToUseReaderWriterLockSlimOverLockInCSharp/WhenToUseReaderWriterLockSlimOverLockInCSharp/LockReadWrite.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+
+namespace WhenToUseReaderWriterLockSlimOverLockInCSharp
+{
+    public static class LockReadWrite
+    {
+        private static readonly Stopwatch stopwatch = new();
+        private static readonly object lockObject = new();
+
+        public static readonly List<int> ListOfNumbers = new();
+
+        public static long Execute(Configuration config)
+        {
+            var threads = new List<Thread>();
+
+            for (var cnt = 0; cnt < config.readerThreadsCount; cnt++)
+            {
+                var readThread = new Thread(ReadListCount);
+                threads.Add(readThread);
+            }
+
+            for (var cnt = 0; cnt < config.writerThreadsCount; cnt++)
+            {
+                var writeThread = new Thread(AddNumbersToList);
+                threads.Add(writeThread);
+            }
+
+            stopwatch.Reset();
+            stopwatch.Start();
+
+            foreach (var thread in threads)
+            {
+                thread.Start(config);
+            }
+
+            foreach (var thread in threads)
+            {
+                thread.Join();
+            }
+
+            stopwatch.Stop();
+
+            return stopwatch.ElapsedMilliseconds;
+        }
```

Comment:
Introduced now the base class. Also, my article mentions threads a lot, so I think keeping them in the example implementation is better. But I agree with your observation about using Tasks otherwise. Would you agree with that?